tests/functionaltests/Tests/NSURLConnectionTests.mm

```diff
@@ -19,6 +19,96 @@
 #import <atomic>
 #import <map>
 #import <random>
+#import <memory>
+
+@interface THRunLoopSpinner : NSObject {
+@public
+    std::shared_ptr<std::atomic<bool>> _shouldStop;
+    BOOL _threadStarted;
+    NSCondition* _threadStartedCondition;
+    NSRunLoop* _loop;
+    pthread_t _thread;
+}
+@property (nonatomic, readonly, retain) NSRunLoop* loop;
+@end
+
+@implementation THRunLoopSpinner
+static void* _helperThreadBody(void* context) {
+    __unsafe_unretained THRunLoopSpinner* unsafeHelper = (__bridge id)context;
```

Comment:
Also, this happens immediately when the thread starts; if the object is dead by then (which it can't be, since we don't return from `init` until the thread is started successfully), something is _terribly wrong_ and we should crash anyway.